{"name": "상품 검색 후 카테고리 버튼들이 동작 하는지 확인", "status": "failed", "statusDetails": {"message": "Failed: ❌ '추천순' 클릭 실패: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//a[contains(text(),'추천순')]\"}\n  (Session info: chrome=135.0.7049.84); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n#0 0x556b2cc3acea <unknown>\n#1 0x556b2c6eb5f0 <unknown>\n#2 0x556b2c73ca33 <unknown>\n#3 0x556b2c73cc21 <unknown>\n#4 0x556b2c78b274 <unknown>\n#5 0x556b2c76268d <unknown>\n#6 0x556b2c788660 <unknown>\n#7 0x556b2c762433 <unknown>\n#8 0x556b2c72eea3 <unknown>\n#9 0x556b2c72fb01 <unknown>\n#10 0x556b2cbffb3b <unknown>\n#11 0x556b2cc03a21 <unknown>\n#12 0x556b2cbe6c32 <unknown>\n#13 0x556b2cc04594 <unknown>\n#14 0x556b2cbcaeef <unknown>\n#15 0x556b2cc28d98 <unknown>\n#16 0x556b2cc28f76 <unknown>\n#17 0x556b2cc39b36 <unknown>\n#18 0x7f112909caa4 <unknown>\n#19 0x7f1129129c3c <unknown>", "trace": "category_name = '추천순', screenshot_name = '추천순'\n\n    def click_category(category_name, screenshot_name):\n        try:\n>           category_button = driver.find_element(By.XPATH,f\"//a[contains(text(),'{category_name}')]\")\n\nsrc/tests/ui_tests/test_ui_category.py:28: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nvenv/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:770: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\nvenv/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:384: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7f158844df10>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"no such element: Unable to locate element: {\\...\\\\n#17 0x556b2cc39b36 \\\\u003Cunknown>\\\\n#18 0x7f112909caa4 \\\\u003Cunknown>\\\\n#19 0x7f1129129c3c \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//a[contains(text(),'추천순')]\"}\nE         (Session info: chrome=135.0.7049.84); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       #0 0x556b2cc3acea <unknown>\nE       #1 0x556b2c6eb5f0 <unknown>\nE       #2 0x556b2c73ca33 <unknown>\nE       #3 0x556b2c73cc21 <unknown>\nE       #4 0x556b2c78b274 <unknown>\nE       #5 0x556b2c76268d <unknown>\nE       #6 0x556b2c788660 <unknown>\nE       #7 0x556b2c762433 <unknown>\nE       #8 0x556b2c72eea3 <unknown>\nE       #9 0x556b2c72fb01 <unknown>\nE       #10 0x556b2cbffb3b <unknown>\nE       #11 0x556b2cc03a21 <unknown>\nE       #12 0x556b2cbe6c32 <unknown>\nE       #13 0x556b2cc04594 <unknown>\nE       #14 0x556b2cbcaeef <unknown>\nE       #15 0x556b2cc28d98 <unknown>\nE       #16 0x556b2cc28f76 <unknown>\nE       #17 0x556b2cc39b36 <unknown>\nE       #18 0x7f112909caa4 <unknown>\nE       #19 0x7f1129129c3c <unknown>\n\nvenv/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py:232: NoSuchElementException\n\nDuring handling of the above exception, another exception occurred:\n\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"d677526e91ec775e85349cd06ac047b3\")>\n\n    @allure.feature(\"UI 테스트\")\n    @allure.story(\"카테고리 기능 테스트\")\n    @allure.title(\"상품 검색 후 카테고리 버튼들이 동작 하는지 확인\")\n    def test_category(driver):\n        driver.get(\"https://www.kurly.com/main\")\n        driver.maximize_window()\n    \n        try:\n          driver.get(\"https://www.kurly.com/main\")\n          time.sleep(4)\n    \n          #검색\n          search_box = driver.find_element(By.XPATH, \"//input[@placeholder='검색어를 입력해주세요']\")\n          search_box.send_keys(\"제로콜라\")\n          search_box.send_keys(Keys.RETURN)\n          time.sleep(4)\n    \n    \n          def click_category(category_name, screenshot_name):\n              try:\n                  category_button = driver.find_element(By.XPATH,f\"//a[contains(text(),'{category_name}')]\")\n                  category_button.click()\n                  time.sleep(4)\n                  capture_screenshot(driver, screenshot_name, \"screenshots_category\")\n              except Exception as e:\n                  capture_screenshot(driver,f\"{screenshot_name}_실패\",\"screenshots_category\")\n                  allure.attach(driver.get_screenshot_as_png(), name=f\"{screenshot_name}_실패\",\n                                attachment_type=allure.attachment_type.PNG)\n                  pytest.fail(f\"❌ '{category_name}' 클릭 실패: {str(e)}\")\n    \n          categories =[\n              (\"추천순\", \"추천순\"),\n              (\"낮은 가격순\", \"낮은가격순\"),\n              (\"높은 가격순\", \"높은가격순\"),\n              (\"판매량순\", \"판매량순\"),\n              (\"혜택순\", \"혜택순\"),\n              (\"신상품순\", \"신상품순\"),\n          ]\n          for category_name, screenshot_name in categories:\n>             click_category(category_name, screenshot_name)\n\nsrc/tests/ui_tests/test_ui_category.py:47: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ncategory_name = '추천순', screenshot_name = '추천순'\n\n    def click_category(category_name, screenshot_name):\n        try:\n            category_button = driver.find_element(By.XPATH,f\"//a[contains(text(),'{category_name}')]\")\n            category_button.click()\n            time.sleep(4)\n            capture_screenshot(driver, screenshot_name, \"screenshots_category\")\n        except Exception as e:\n            capture_screenshot(driver,f\"{screenshot_name}_실패\",\"screenshots_category\")\n            allure.attach(driver.get_screenshot_as_png(), name=f\"{screenshot_name}_실패\",\n                          attachment_type=allure.attachment_type.PNG)\n>           pytest.fail(f\"❌ '{category_name}' 클릭 실패: {str(e)}\")\nE           Failed: ❌ '추천순' 클릭 실패: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//a[contains(text(),'추천순')]\"}\nE             (Session info: chrome=135.0.7049.84); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE           Stacktrace:\nE           #0 0x556b2cc3acea <unknown>\nE           #1 0x556b2c6eb5f0 <unknown>\nE           #2 0x556b2c73ca33 <unknown>\nE           #3 0x556b2c73cc21 <unknown>\nE           #4 0x556b2c78b274 <unknown>\nE           #5 0x556b2c76268d <unknown>\nE           #6 0x556b2c788660 <unknown>\nE           #7 0x556b2c762433 <unknown>\nE           #8 0x556b2c72eea3 <unknown>\nE           #9 0x556b2c72fb01 <unknown>\nE           #10 0x556b2cbffb3b <unknown>\nE           #11 0x556b2cc03a21 <unknown>\nE           #12 0x556b2cbe6c32 <unknown>\nE           #13 0x556b2cc04594 <unknown>\nE           #14 0x556b2cbcaeef <unknown>\nE           #15 0x556b2cc28d98 <unknown>\nE           #16 0x556b2cc28f76 <unknown>\nE           #17 0x556b2cc39b36 <unknown>\nE           #18 0x7f112909caa4 <unknown>\nE           #19 0x7f1129129c3c <unknown>\n\nsrc/tests/ui_tests/test_ui_category.py:36: Failed"}, "attachments": [{"name": "추천순_실패", "source": "34489c0f-bc96-41e7-918d-887256cd2c1f-attachment.png", "type": "image/png"}, {"name": "Failure Screenshot", "source": "f5438abb-503a-4dc7-85e3-91e1e2558f56-attachment.png", "type": "image/png"}, {"name": "Failure Screenshot", "source": "0294128b-62b1-4c80-8cb7-ea46322749fb-attachment.png", "type": "image/png"}, {"name": "stdout", "source": "6a996052-f2f5-4af7-be18-6f84d023a6d6-attachment.txt", "type": "text/plain"}], "start": 1745287131493, "stop": 1745287143441, "uuid": "4c771eb1-5f61-4828-b79b-ae332d5ba47e", "historyId": "bdd47d824380b8945b6f725c5ea0b675", "testCaseId": "bdd47d824380b8945b6f725c5ea0b675", "fullName": "src.tests.ui_tests.test_ui_category#test_category", "labels": [{"name": "feature", "value": "UI 테스트"}, {"name": "story", "value": "카테고리 기능 테스트"}, {"name": "parentSuite", "value": "src.tests.ui_tests"}, {"name": "suite", "value": "test_ui_category"}, {"name": "host", "value": "fv-az2027-756"}, {"name": "thread", "value": "2582-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "src.tests.ui_tests.test_ui_category"}]}